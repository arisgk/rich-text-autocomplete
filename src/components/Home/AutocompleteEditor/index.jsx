import React, { Component } from 'react';
import { CompositeDecorator, EditorState } from 'draft-js';
import Editor from 'draft-js-plugins-editor';
import 'draft-js-mention-plugin/lib/plugin.css';
import createMentionPlugin, { defaultSuggestionsFilter } from 'draft-js-mention-plugin';
import mentions from '../../../data/mentions';

const mentionPlugin = createMentionPlugin();
const { MentionSuggestions } = mentionPlugin;
const plugins = [mentionPlugin];

const styles = {
  root: {
    padding: 20,
    width: 600,
  },
  editor: {
    border: '1px solid #ddd',
    cursor: 'text',
    fontSize: 16,
    minHeight: 40,
    padding: 10,
  },
};

class AutocompleteEditor extends Component {
  constructor() {
    super();

    this.state = {
      editorState: EditorState.createEmpty(),
      suggestions: mentions,
    };

    this.focus = () => this.refs.editor.focus();
    this.onChange = editorState => this.setState({ editorState });
    this.onSearchChange = this.onSearchChange.bind(this);
  }

  onSearchChange({ value }) {
    this.setState({
      suggestions: defaultSuggestionsFilter(value, mentions),
    });
  }

  render() {
    return (
      <div style={styles.root}>
        <div style={styles.editor} onClick={this.focus}>
          <Editor
            editorState={this.state.editorState}
            onChange={this.onChange}
            ref="editor"
            plugins={plugins}
            spellCheck
          />
          <MentionSuggestions
            onSearchChange={this.onSearchChange}
            suggestions={this.state.suggestions}
            onAddMention={this.onAddMention}
          />
        </div>
      </div>
    );
  }
}

export default AutocompleteEditor;
