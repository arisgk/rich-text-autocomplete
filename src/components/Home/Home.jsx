import React from 'react';
import Radium from 'radium';
import AppBar from 'material-ui/AppBar';
import AutocompleteEditorContainer from '../../containers/AutocompleteEditorContainer';

const styles = {
  container: {
    height: '100vh',
    display: 'flex',
    flexDirection: 'column',
  },
  main: {
    flex: 'auto',
    display: 'flex',
    WebkitAlignItems: 'center',
    AlignItems: 'center',
    WebkitJustifyContent: 'center',
    JustifyContent: 'center',
    WebkitFlexDirection: 'column',
    flexDirection: 'column',
    paddingBottom: '10%',
  },
  h1: {
    fontSize: '140%',
    color: 'rgba(0,0,0,0.87)',
    marginBottom: 24,
  },
};

const Home = () => (
  <div style={styles.container}>
    <AppBar title="Autocomplete" />
    <div style={styles.main}>
      <h1 style={styles.h1}>Editor</h1>
      <AutocompleteEditorContainer />
    </div>
  </div>
);

export default Radium(Home);
