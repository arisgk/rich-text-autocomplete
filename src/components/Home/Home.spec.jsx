/* global describe, it, expect */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Home from './Home';

describe('Home', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Home />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
