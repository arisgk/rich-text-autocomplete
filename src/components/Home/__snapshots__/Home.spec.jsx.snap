// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100vh",
    }
  }
>
  <AppBar
    showMenuIconButton={true}
    title="Autocomplete"
    zDepth={1}
  />
  <div
    data-radium={true}
    style={
      Object {
        "AlignItems": "center",
        "JustifyContent": "center",
        "WebkitAlignItems": "center",
        "WebkitFlexDirection": "column",
        "WebkitJustifyContent": "center",
        "display": "flex",
        "flex": "auto",
        "flexDirection": "column",
        "paddingBottom": "10%",
      }
    }
  >
    <h1
      data-radium={true}
      style={
        Object {
          "color": "rgba(0,0,0,0.87)",
          "fontSize": "140%",
          "marginBottom": 24,
        }
      }
    >
      Editor
    </h1>
    <Connect(AutocompleteEditor) />
  </div>
</div>
`;
